ipn/ipnlocal/local.go

```diff
@@ -425,6 +425,13 @@ func (b *LocalBackend) updateStatus(sb *ipnstate.StatusBuilder, extraLocked func
 			s.CurrentTailnet.MagicDNSEnabled = b.netMap.DNS.Proxied
 			s.CurrentTailnet.Name = b.netMap.Domain
 		}
+		if b.prefs != nil && !b.prefs.ExitNodeID.IsZero() {
+			s.ExitNodeStatus = &ipnstate.ExitNodeStatus{
+				ID:           b.prefs.ExitNodeID,
+				Online:       b.exitNodeOnline(),
+				TailscaleIPS: b.exitNodeIPS(),
```

Comment:
these are both linear scans over all peers and both assume methods below look like they're safe to use anywhere but aren't documented with the `Locked` suffix and don't check that `b.netMap` is nil.

How about instead you add a method to to the `b.netMap` type like `PeerWithStableID` so you only do the linear search once (and later we can optimize it into a map; I think such a TODO already exists somewhere) and then this code is like:

```go
     if b.prefs != nil {
              if exitPeer, ok := b.netMap.PeerWithStableID(b.prefs.ExitNodeID); ok {
                  s.ExitNodeStatus = &ipnstate.ExitNodeStatus{
                       ....
```